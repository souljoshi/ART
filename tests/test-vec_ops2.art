int main()
{
	vec v1;
	vec v2;
	vec v3;
	vec v4;

    // scalar multiplication
	v1 = 2.0 * <1.0, 55.1>;
    v2 = 2 * <1.0, 55.1>; // integer promotion
    v3 = <1.0, 55.1> * 2.0;
    v4 = <1.0, 55.1> * 2;

    printf(v1[0]);
    printf(v1[1]);
    printf(v2[0]);
    printf(v2[1]);
    printf(v3[0]);
    printf(v3[1]);
    printf(v4[0]);
    printf(v4[1]);

}