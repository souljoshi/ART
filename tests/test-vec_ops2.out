2.000000
110.200000
2.000000
110.200000
2.000000
110.200000
2.000000
110.200000
